Added a bunch of new config options

boerdereinar opened the pull request:
Hi there. Wanted to let you know that I love this mod and just wanted more settings to fiddle with and to get this to work on a server that is always underground and in an ocean biome. So I just decided to make a bunch of settings and if you like what I did you can maybe merge it with the main branch. If you run into any issues with this branch or have any suggestions, do let me know and I'll see what I can do.

Here's everything I've changed:

Added:
- Spawn rate settings for eyes
- Spawn rate settings for will o' wisps
- Spawn rate settings for chorus petals
- Option for fireflies to spawn every time of day
- Option for fireflies to spawn underground
- New biome settings tab
- Firefly/glowworm/plankton spawn rate settings for each biome category
- New aura settings tab
- Spawn rate/delay settings for each aura

Changed:
- mod menu api from io.github.prospector to com.terraformersmc to support future versions of minecraft
- saving the settings file only happens once after pressing save & quit in mod menu
- settings file is now sorted alphabetically
- changed some data classes to records

doctor4t commented:
Seems like a very interesting contribution! I'll take a look

doctor4t commented:
After testing a bit, it seems like the "Fireflies always spawn" option doesn't work, because you forgot to make it so fireflies don't die during daylight when that option is enabled.

A thing I'm not really willing to include for now though are the aura configuration options, as I'm planning to make it so individual donators could customize their aura spawn rates, maybe as part of a higher tier, so for now I'd comment these aura config options out.

Great PR! I'll let you fix the fireflies dying during daylight even with the "Fireflies always spawn" option and then I'll merge it.

boerdereinar commented:
Thanks for checking out my branch. I fixed the issue with fireflies immediately despawning and also fixed that the fireflies didn't work in the nether and end. I also removed plankton and glowworms from the nether and glowworms fromm the end. I also commented out all the aura code so if you want to restore/remove it, you can do it any time. I haven't found any more issues so if you think it's good now, you can go ahead and merge.

doctor4t commented:
Sounds good, thanks a lot for the valuable contribution! Will push out a new update soon, thanks to your work.
